src/main/java/build/buildfarm/instance/shard/ShardActionCache.java

```diff
@@ -76,11 +82,11 @@ public void put(ActionKey actionKey, ActionResult actionResult) {
 
   @Override
   public void invalidate(ActionKey actionKey) {
-    actionResultCache.invalidate(actionKey);
+    actionResultCache.synchronous().invalidate(actionKey);
   }
 
   @Override
   public void readThrough(ActionKey actionKey, ActionResult actionResult) {
```

Comment:
I still question this name, maybe I'm off base here.

Reply:
It seemed okay to me.  we can simplify things more when framework is swapped.